src/main/java/appeng/datagen/providers/recipes/AE2RecipeProvider.java

```diff
@@ -49,6 +49,9 @@ public static JsonObject toJson(ItemStack stack) {
         if (stack.getCount() > 1) {
             stackObj.addProperty("count", stack.getCount());
         }
+        if (stack.hasTag()) {
+            stackObj.addProperty("data", stack.getTag().toString());
+        }
```

Comment:
Does that even work in Vanilla? Just for the output, I presume?

Reply:
This is to save the NBT data _to_ the JSON, however in the meantime it occurred to me that I hadn't checked whether or not this could affect any other AE2 recipes. Might have to do some more on it anyways.